src/timebar/charts/activity.js

```diff
@@ -1,42 +1,64 @@
-import React, { Component } from 'react'
+import React, { useContext, useMemo } from 'react'
 import PropTypes from 'prop-types'
 import { area, curveStepAfter } from 'd3-shape'
-import { animated, Spring } from 'react-spring/renderprops'
 import { getTime } from '../utils'
 import ImmediateContext from '../immediateContext'
+import { DEFAULT_CSS_TRANSITION } from '../constants'
 import styles from './activity.module.css'
 
 const TOP_MARGIN = 5
 const BOTTOM_MARGIN = 20
 
-class Activity extends Component {
-  static contextType = ImmediateContext
-  render() {
-    const { activity, absoluteEnd, outerScale, outerWidth, graphHeight } = this.props
-    const { immediate } = this.context
-
-    // because data stops at last day midnight, add an extra day with the same data
-    // to allow the curve to go to the end of the day
-    const lastDay = activity[activity.length - 1]
-    const addedLastDay = { date: getTime(absoluteEnd), value: lastDay.value }
-
-    const finalHeight = graphHeight - TOP_MARGIN - BOTTOM_MARGIN
-    const middle = TOP_MARGIN + finalHeight / 2
-
-    const areaGenerator = area()
-      .x((d) => outerScale(d.date))
-      .y0((d) => middle - (finalHeight * d.value) / 2)
-      .y1((d) => middle + (finalHeight * d.value) / 2)
-      .curve(curveStepAfter)
-
-    return (
-      <svg width={outerWidth} height={graphHeight} className={styles.Activity}>
-        <Spring native immediate={immediate} to={{ d: areaGenerator([...activity, addedLastDay]) }}>
-          {(style) => <animated.path d={style.d} fill="pink" fillOpacity={0.9} />}
-        </Spring>
-      </svg>
-    )
-  }
+const getPath = (graphHeight, activity, absoluteEnd, overallScale) => {
+  const finalHeight = graphHeight - TOP_MARGIN - BOTTOM_MARGIN
+  const middle = TOP_MARGIN + finalHeight / 2
+
+  const areaGenerator = area()
+    .x((d) => overallScale(d.date))
+    .y0((d) => middle - (finalHeight * d.value) / 2)
+    .y1((d) => middle + (finalHeight * d.value) / 2)
+    .curve(curveStepAfter)
+
+  // because data stops at last day midnight, add an extra day with the same data
+  // to allow the curve to go to the end of the day
+  const lastDay = activity[activity.length - 1]
+  const addedLastDay = { date: getTime(absoluteEnd), value: lastDay.value }
+
+  const path = areaGenerator([...activity, addedLastDay])
+
+  return path
+}
+
+const Activity = ({
+  activity,
+  absoluteEnd,
+  innerWidth,
+  graphHeight,
+  cssTransform,
+  overallScale,
+}) => {
+  const { immediate } = useContext(ImmediateContext)
```

Comment:
Are we using this ? if not I'd remove it to avoid unnecessary re-renders

Reply:
Needed it eventually (with last commit), but good catch